### PR TITLE
[IOCOM-1106] Update "node-oidc-provider" to its latest version

### PR DESCRIPTION
**Pull request by lucacavallaro**
Update `node-oidc-provider` from `7.x` to `8.x` (latest version).

### List of changes
1. Since `8.x` is an `ESM-only` package ([reference](https://github.com/panva/node-oidc-provider/releases/tag/v8.0.0#:ESM:~:text=signal%20property%20instead.-,oidc%2Dprovider%20is%20now%20an%20ESM%2Donly%20module,-%2C%20it%20must%20now)) I migrated this project to `ESM` (`"type": "module"`)
2. The migration to ESM required a change on all import (from `import x from "foo"` to `import x from "foo.js"`)
3. It also broke the unit tests, so I replaced `jest` with `vitest`, removed `jest-mock-extended` and wrote new (manual) mocks (`adapters/vitest.ts`)
4. I updated `typescript` (and its `tsconfig`), removed `gulp` and `husky`
5. `.env.default` is now `.env.example` and it serve at reference for environment variables, the new file for envs in dev mode is `.env` (that is not checked out in this repo)
6. updated `eslint` to its latest version

### CODE REVIEW

**Review by gunzip — APPROVED** [review]
lgtm

**Review by Garma00 — DISMISSED** [review]
lgtm

**Review by Garma00 — DISMISSED** [review]

File: src/adapters/cosmosdb/utils.ts [hunk 1]
```diff
@@ -1,15 +1,18 @@
-// import { constVoid } from "fp-ts/lib/function";
+// import { constVoid } from "fp-ts/lib/function.js";
```
Comment (michaeldisaro) [inline src/adapters/cosmosdb/utils.ts:1]: why not removing this dead code?

File: src/useCases/FindGrantUseCases.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ export const FindGrantUseCase =
       TE.map(RA.head),
       TE.fold(
         (error) => {
-          logger.error("Some error during FindGrantUseCase", error.causedBy);
+          logger.error("Some error during FindGrantUseCase.js", error.causedBy);
```
Comment (michaeldisaro) [inline src/useCases/FindGrantUseCases.ts:33]: .js also in logs?

**Review by michaeldisaro — APPROVED** [review]
Just a couple of minor comments.

**Review by michaeldisaro — APPROVED** [review]

**Pull request merged**